packages/react-query-devtools/src/index.ts

```diff
@@ -1 +1,16 @@
-export * from './devtools'
+import * as devtools from './devtools'
+
+export let ReactQueryDevtools: typeof devtools['ReactQueryDevtools'] =
+  function () {
+    return null
+  }
+
+export let ReactQueryDevtoolsPanel: typeof devtools['ReactQueryDevtoolsPanel'] =
+  function () {
+    return null
+  } as any
+
+if (process.env.NODE_ENV !== 'production') {
```

Comment:
I believe this will cause issues when the environment is 'test'. 

Perhaps something like this?

`!['production', 'test'].includes(process.env.NODE_ENV)`

Reply:
`!['production', 'test'].includes(process.env.NODE_ENV)` may not work since minifier needs simple logic to do DCE. `process.env.NODE_ENV !== 'production' || process.env.NODE_ENV !== 'test'` may work but it's usually better to bundle in `production` mode when running e2e tests to avoid development related stuff in test.

Many other tools use `process.env.NODE_ENV !== 'production'` (e.g. redux: https://github.com/reduxjs/redux-devtools/blob/a8883f287de2e36a12a4058635f8474cda34a0d4/packages/redux-devtools-extension/src/logOnlyInProduction.ts#L22) so a custom env will leave mix of dev and production code in the bundle.

Reply:
So, using 4.0.5 I am seeing this issue because react-testing-library uses `jsdom` which is currently failing because `window.matchMedia` is not supported:
<img width="911" alt="Screenshot 2022-07-20 at 11 54 39" src="https://user-images.githubusercontent.com/1246923/179965433-f9d2e33c-5050-46f7-b57d-b2ffd7e671c0.png">

Reply:
the latest attempt doesn't use an env check at all

Reply:
okay I think I got it now - added the !== development env check again for commonjs builds, and esm builds now use the separate `exports` field in package.json